Runtimes/Supplemental/Differentiation/CMakeLists.txt

```
@@ -0,0 +1,96 @@
+
+include(gyb)
+include(AvailabilityMacros)
+include(CatalystSupport)
+include(EmitSwiftInterface)
+include(ResourceEmbedding)
+
+option(${PROJECT_NAME}_ENABLE_VECTOR_TYPES "Enable vector support"
+  ${SwiftCore_ENABLE_VECTOR_TYPES})
+
+option(${PROJECT_NAME}_ENABLE_LIBRARY_EVOLUTION "Enable library evolution"
+  ${SwiftCore_ENABLE_LIBRARY_EVOLUTION})
+
+add_compile_options(
+  $<$<COMPILE_LANGUAGE:Swift>:-explicit-module-build>
+  $<$<COMPILE_LANGUAGE:Swift>:-nostdlibimport>
+  $<$<COMPILE_LANGUAGE:Swift>:-parse-stdlib>
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-library-level api>"
+  $<$<COMPILE_LANGUAGE:Swift>:-enforce-exclusivity=unchecked>
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-target-min-inlining-version min>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature NoncopyableGenerics2>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature SuppressedAssociatedTypes>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature SE427NoInferenceOnExtension>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature NonescapableTypes>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature LifetimeDependence>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature MemberImportVisibility>"
+  $<$<AND:$<BOOL:${SwiftDifferentiation_ENABLE_LIBRARY_EVOLUTION}>,$<COMPILE_LANGUAGE:Swift>>:-enable-library-evolution>)
+
+if(SwiftDifferentiation_ENABLE_VECTOR_TYPES)
+  gyb_expand(SIMDDifferentiation.swift.gyb SIMDDifferentiation.swift)
+endif()
+gyb_expand(TgmathDerivatives.swift.gyb TgmathDerivatives.swift)
+gyb_expand(FloatingPointDifferentiation.swift.gyb
+  FloatingPointDifferentiation.swift)
+
+add_library(swift_Differentiation
+  AnyDifferentiable.swift
+  ArrayDifferentiation.swift
+  Differentiable.swift
+  DifferentialOperators.swift
+  DifferentiationUtilities.swift
+  OptionalDifferentiation.swift
+  $<$<BOOL:${SwiftDifferentiation_ENABLE_VECTOR_TYPES}>:SIMDDifferentiation.swift>
+  TgmathDerivatives.swift
+  FloatingPointDifferentiation.swift
+  linker-support/magic-symbols-for-install-name.c)
+
+set_target_properties(swift_Differentiation PROPERTIES
+  Swift_MODULE_NAME _Differentiation)
+
+if(APPLE AND BUILD_SHARED_LIBS)
+  target_link_options(swift_Differentiation PRIVATE "SHELL:-Xlinker -headerpad_max_install_names")
+endif()
+
+target_link_libraries(swift_Differentiation PRIVATE swiftCore SwiftMath)
+
+set(${PROJECT_NAME}_INSTALL_LIBDIR "${CMAKE_INSTALL_LIBDIR}/swift$<$<NOT:$<BOOL:${BUILD_SHARED_LIBS}>>:_static>$<$<BOOL:${Supplemental_INSTALL_NESTED_SUBDIR}>:/${Supplemental_PLATFORM_SUBDIR}/${Supplemental_ARCH_SUBDIR}>" CACHE STRING "")
+set(${PROJECT_NAME}_INSTALL_SWIFTMODULEDIR "${CMAKE_INSTALL_LIBDIR}/swift$<$<NOT:$<BOOL:${BUILD_SHARED_LIBS}>>:_static>$<$<BOOL:${Supplemental_INSTALL_NESTED_SUBDIR}>:/${Supplemental_PLATFORM_SUBDIR}>" CACHE STRING "")
+
+install(TARGETS swift_Differentiation
+  EXPORT SwiftSupplementalTargets
+  COMPONENT SwiftCore_runtime
```

I would suggest to match the same name we are using in #81310

```suggestion
  COMPONENT ${PROJECT_NAME}_runtime
```